src/EventProducer.cc

```diff
@@ -59,13 +167,91 @@ void EventProducer::produce(edm::Event& event_, const edm::EventSetup& eventSetu
     if (event_.getByToken(m_lhe_info_token, lhe_info)) {
         lhe_originalXWGTUP = lhe_info->originalXWGTUP();
         lhe_SCALUP = lhe_info->hepeup().SCALUP;
+
+        // Scale variations
+        for (auto& m: m_scale_variations_matching) {
+            float weight = lhe_info->weights()[m.second].wgt / lhe_originalXWGTUP;
+            scale_weights.push_back(weight);
+        }
+
+        // PDF variations
+
         for (auto& weight: lhe_info->weights()) {
             lhe_weights.push_back(std::make_pair(weight.id, weight.wgt));
+#ifdef DEBUG_PDF
+            std::cout << weight.id << " = " << weight.wgt << " (" << weight.wgt / lhe_originalXWGTUP << ")" << std::endl;
+#endif
         }
+
+        size_t from = 0;
+        size_t to = m_pdf_weights_matching.size();
+
+        if (has_alphas_uncertainty)
+            to -= 2;
+
+        size_t n = to - from;
+
+        // Replicas. Compute mean and RMS of the distribution
+        float mean = 0.;
+        for (size_t i = from; i < to; i++) {
+            float weight = lhe_info->weights()[m_pdf_weights_matching[i].second].wgt / lhe_originalXWGTUP;
+            mean += weight;
+        }
+        mean /= n;
+
+        // Compute standard deviation
+        float rms = 0.;
+        for (size_t i = from; i < to; i++) {
+            float weight = lhe_info->weights()[m_pdf_weights_matching[i].second].wgt / lhe_originalXWGTUP;
+            rms += (weight - mean) * (weight - mean);
+        }
+        rms = std::sqrt(1 / (n - 1.) * rms);
+
+#ifdef DEBUG_PDF
+        std::cout << "PDF uncertainty: " << rms << " (mean value: " << mean << ")" << std::endl;
+#endif
+
+        // Add alphaS uncertainty
+        if (has_alphas_uncertainty) {
+            float weight_alphas_up = lhe_info->weights()[m_pdf_weights_matching[n].second].wgt / lhe_originalXWGTUP;
+            float weight_alphas_down = lhe_info->weights()[m_pdf_weights_matching[n + 1].second].wgt / lhe_originalXWGTUP;
+            // 0.75 factor is needed because NNPDF30 alphaS uncertainties are
+            // +- 0.002 whereas recommendations for Run2 are +- 0.0015
+            float alphas_uncertainty = 0.75 * (weight_alphas_up - weight_alphas_down) / 2.;
+#ifdef DEBUG_PDF
+            std::cout << "alphaS uncertainty: " << alphas_uncertainty << std::endl;
+#endif
+            rms = std::sqrt(rms * rms + alphas_uncertainty * alphas_uncertainty);
```

Comment:
Indeed :) If preferred, we can create a new branch for alphaS and one for the combination, but IMO it's useless since you won't apply one without the other.